### PR TITLE
feat: [IA-406] Fixed iOS accessibility internal translations

### PR DESCRIPTION
**Pull request by dgopsq**
## Short description
This PR is going to patch the React Native NPM package, adding an Italian localisation file to translate various internal strings used for accessibility.

The patch will just **add** a new file inside the React Native module, not changing anything else.

For reference see [this blog post](https://valtterilaine.medium.com/react-native-accessibilityrole-i18n-support-for-ios-and-android-16d896e69178).

image

## List of changes proposed in this pull request
- Translated the `checked` and `not checked` values in: `selezionato` and `non selezionato`

## How to test
Using the accessibility inspector to analyse the services configuration's radio buttons while using iOS in Italian should result in having the `checked` and `not checked` values translated as `selezionato` and `non selezionato`.

### CODE REVIEW

**Comment by dgopsq** [comment]
@thisisjp I added you as a reviewer too because there's translated text but not in the usual place (https://github.com/pagopa/io-app/pull/3791/files#diff-5f8f59afee259f80d174c30a791f101d87b9bc1086431df7a82e41345bde40a2R27-R28)

File: patches/react-native+0.64.2.patch [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
++"progressbar"="progress bar";
++"radio"="radio button";
++"radiogroup"="radio group";
++"scrollbar"="scroll bar";
++"spinbutton"="spin button";
++"switch"="switch";
++"tab"="tab description";
++"tablist"="tab list";
++"timer"="timer";
++"toolbar"="tool bar";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:26]:
```suggestion
+"toolbar"="barra degli strumenti";
```

File: patches/react-native+0.64.2.patch [hunk 2]
```diff
@@ -1,14 +1,46 @@
...
++"checked"="selezionato";
++"unchecked"="non selezionato";
++"busy"="busy";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:29]:
```suggestion
+"busy"="occupato";
```

File: patches/react-native+0.64.2.patch [hunk 3]
```diff
@@ -1,14 +1,46 @@
...
++"expanded"="expanded";
++"collapsed"="collapsed";
++"mixed"="mixed";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:32]:
```suggestion
+"mixed"="misto";
```

File: patches/react-native+0.64.2.patch (hunk 2 above)
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:29]:
```suggestion
+"busy"="occupato";
```

File: patches/react-native+0.64.2.patch [hunk 4]
```diff
@@ -1,14 +1,46 @@
...
++"spinbutton"="spin button";
++"switch"="switch";
++"tab"="tab description";
++"tablist"="tab list";
++"timer"="timer";
++"toolbar"="tool bar";
++"checked"="selezionato";
++"unchecked"="non selezionato";
++"busy"="busy";
++"expanded"="expanded";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:30]:
```suggestion
+"expanded"="espanso";
```

File: patches/react-native+0.64.2.patch [hunk 5]
```diff
@@ -1,14 +1,46 @@
...
++"collapsed"="collapsed";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:31]:
```suggestion
+"collapsed"="compresso";
```

File: patches/react-native+0.64.2.patch [hunk 6]
```diff
@@ -1,14 +1,46 @@
...
+index 0000000..f09950f
+--- /dev/null
++++ b/node_modules/react-native/React/AccessibilityResources/it.lproj/Localizable.strings
+@@ -0,0 +1,26 @@
++/*
++  Localizable.strings
++  React
++*/
++"alert"="alert";
++"checkbox"="checkbox";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:12]:
```suggestion
+"checkbox"="casella di controllo";
```

File: patches/react-native+0.64.2.patch [hunk 7]
```diff
@@ -1,14 +1,46 @@
...
++"combobox"="combo box";
++"menu"="menu";
++"menubar"="menu bar";
++"menuitem"="menu item";
++"progressbar"="progress bar";
++"radio"="radio button";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:18]:
```suggestion
+"radio"="pulsante di scelta";
```

File: patches/react-native+0.64.2.patch [hunk 8]
```diff
@@ -1,14 +1,46 @@
...
++  Localizable.strings
++  React
++*/
++"alert"="alert";
++"checkbox"="checkbox";
++"combobox"="combo box";
++"menu"="menu";
++"menubar"="menu bar";
++"menuitem"="menu item";
++"progressbar"="progress bar";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:17]:
```suggestion
+"progressbar"="barra di avanzamento";
```

File: patches/react-native+0.64.2.patch [hunk 9]
```diff
@@ -1,14 +1,46 @@
...
++"radio"="radio button";
++"radiogroup"="radio group";
++"scrollbar"="scroll bar";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:20]:
```suggestion
+"scrollbar"="barra di scorrimento";
```

File: patches/react-native+0.64.2.patch [hunk 10]
```diff
@@ -1,14 +1,46 @@
...
++/*
++  Localizable.strings
++  React
++*/
++"alert"="alert";
++"checkbox"="checkbox";
++"combobox"="combo box";
++"menu"="menu";
++"menubar"="menu bar";
++"menuitem"="menu item";
```
Comment (thisisjp) [inline patches/react-native+0.64.2.patch:16]:
```suggestion
+"menuitem"="voce del menu";
```

**Review by thisisjp — COMMENTED** [review]
@dgopsq I noticed there are other UI elements which required translation... I didn't go through all of them, but better than nothing, right? :)

**Review by fabriziofff — CHANGES_REQUESTED** [review]
Could you update the https://github.com/pagopa/io-app/blob/master/patches/patches.md#react-native0642 file, in order to track the reason of this patch?

**Comment by dgopsq** [comment]
Thank you @thisisjp, I integrated everything 🙂 Also changed the PR title to a more generic one since we're going to address more translations than the ones in the ticket!

@fabriziofff I just added a small changelog, with a reference to this issue!

**Review by fabriziofff — APPROVED** [review]

**Pull request merged**